File: figuras/Python/figuras/Cuadrado.py
'''
Created on 5 abr. 2019



@author: Francisco Javier González Sabariego
'''
from figuras.Rectangulo import Rectangulo

class Cuadrado(Rectangulo):
    def __init__(self, lado):
        Rectangulo.__init__(self, lado, lado)
    
    
    def __eq__(self, other):
        '''
        Método equals, compara dos objetos
        '''
        if self.ancho == other.ancho:
            return True
        else:
            return False
    
    
    def __gt__(self, other):
        '''
        Método greater than, retorna verdadero si el objeto (self) que llama al método
        es mayor que el objeto se le pasa como parámetro, de lo contrario devuelve falso.
        '''
        if self.ancho > other.ancho:
            return True
        else:
            return False
    
    
    def __ge__(self, other):
        '''
        Método greater than or equals to, retorna verdadero si el objeto (self) que llama al método
        es mayor o igual que el objeto se le pasa como parámetro, de lo contrario devuelve falso.
        '''
        if self.ancho > other.ancho or self.ancho == other.ancho:
            return True
        else:
            return False
    
    
    def __lt__(self, other):
        '''
        Método less than, retorna verdadero si el objeto (self) que llama al método
        es menor que el objeto se le pasa como parámetro, de lo contrario devuelve falso.
        '''
        if not self.__gt__(other):
            return True
        else:
            return False
    
    
    def __le__(self, other):
        '''
        Método less than or equals to, retorna verdadero si el objeto (self) que llama al método
        es menor que el objeto se le pasa como parámetro, de lo contrario devuelve falso.
        '''
        if self.ancho < other.ancho or self.ancho == other.ancho:
            return True
        else:
            return False
    
    
    @property
    def lado(self):
        return self.__ancho
    
    
    @lado.setter
    def lado(self, l):
        if l<1 or l>10:
            raise ArithmeticError
        else:
            self.__ancho=l 

File: figuras/Java/src/figuras/Cuadrado.java
package figuras;

/**
 * Clase Cuadrado, hereda de la Clase Rectángulo e implementa
 * la interfaz Comparable.
 * 
 * Esta clase crea una figura geométrica de tipo cuadrado, se
 * le pasa como parámetro al constructor el valor del lado.
 * 
 * @author Francisco Javier González Sabariego
 * @version 1.0 || 05/04/2019
 *
 */
public class Cuadrado extends Rectangulo implements Comparable<Cuadrado>{
  
  /**
   * Constructor.
   * 
   * @param lado Valor (int) del lado del cuadrado.
   */
  public Cuadrado (int lado) {
    super(lado, lado);
  }
  
  /**
   * Compara dos cuadrados entre sí.
   * 
   * @return si son iguales devuelve 0 y otro valor si no lo son.
   */
  @Override
  public int compareTo(Cuadrado c) {
    return this.getAncho()-c.getAncho(); 
  }
  /**
   * Asigna un valor al lado
   * 
   * @param lado  Valor (int) del lado.
   */
  public void setLado(int lado) {
    if (lado>0 && lado<11) {
      setAncho(lado);
      setAlto(lado);
    } else {
      throw new ArithmeticException();
    }      
  }
  
  /**
   * Devuelve el valor del lado.
   * 
   * @return Lado (int) del cuadrado.
   */
  public int getLado() {
    return this.getAncho();
  }
  
}
